core/views/GearViews.py

```diff
@@ -1,26 +1,30 @@
-from django.http import HttpResponse
 from django.views.generic.detail import DetailView
 
-from django.shortcuts import render
 from django.utils import timezone
 
 from core.models.GearModels import Gear
+from core.views.ViewList import RestrictedViewList
+from core.views.common import ModelDetailView
 
 
-class GearDetailView(DetailView):
+class GearDetailView(ModelDetailView):
 
     model = Gear
 
-    def get_context_data(self, **kwargs):
-        context = super().get_context_data(**kwargs)
-        context['now'] = timezone.now()
-        context['opts'] = self.model._meta
-        context['app_label'] = self.model._meta.app_label
-        context['change'] = False
-        context['is_popup'] = False
-        context['add'] = False
-        context['save_as'] = False
-        context['has_delete_permission'] = False
-        context['has_change_permission'] = False
-        context['has_add_permission'] = False
-        return context
+
+class GearViewList(RestrictedViewList):
+
+    def test_func(self):
+        """Users can see the gear list if the have that permission"""
+        return self.request.user.has_permission("view_gear")
+
+    def can_view_all(self):
+        """If a user isn't allowed to view all gear, only show the gear that is available"""
+        return self.request.user.has_permission("view_all_gear")
+
+    def set_restriction_filters(self):
+        """Staffers should be able to see active gear, members should only see available gear"""
+        if self.request.user.is_staffer:
+            self.restriction_filters["status__lte"] = 3
```

Comment:
What is `lte`?

Reply:
The way django interprets filters in a dictionary is as follows:

Key:
  - Position 1: field name that is being filtered
  - Position 2: double underscore (__) is treated as a separator
  - Position 3: operator, in this case lte is 'less than or equal to'
Value:
  - the 'value' used in this expression, which is generated from the key

The returned subset of the queryset is all objects for which the above expression returns true.
In this case, all object where: object.status >= 3


for more info see:
https://docs.djangoproject.com/en/2.1/topics/db/queries/#field-lookups